kp_test.go

```diff
@@ -5253,3 +5253,277 @@ func TestListKeyFilter(t *testing.T) {
 	assert.Contains(t, err.Error(), "INVALID_QUERY_PARAM_ERR")
 	assert.True(t, gock.IsDone(), "Expected HTTP requests not called")
 }
+
+func TestKMIPMgmtAPI(t *testing.T) {
+	defer gock.Off()
+	UUID := "feddecaf-0000-0000-0000-1234567890ab"
+	exampleIBMID := "IBMid-0000000000"
+	kmipAdapterProfile := "native_1.0"
+	crkUUID := "beddecaf-0000-0000-0000-1234567890ab"
+	timestamp := time.Now()
+	newAdapter := KMIPAdapter{
+		ID:      UUID,
+		Name:    "kmip-adapter-123",
+		Profile: kmipAdapterProfile,
+		ProfileData: map[string]string{
+			"crk_id": crkUUID,
+		},
+		Description: "our 123rd kmip adapter",
+		CreatedAt:   &timestamp,
+		CreatedBy:   exampleIBMID,
+	}
+	newCertificate := KMIPClientCertificate{
+		ID:        UUID,
+		Name:      "kmip-client-certificate",
+		CreatedAt: &timestamp,
+		CreatedBy: exampleIBMID,
+	}
+	newKmipObject := KMIPObject{
+		ID:              UUID,
+		KMIPObjectType:  2,
+		ObjectState:     1,
+		CreatedByCertID: newCertificate.ID,
+		CreatedBy:       exampleIBMID,
+		CreatedAt:       &timestamp,
+	}
+
+	singleAdapter := &KMIPAdapters{
+		Metadata: KeysMetadata{
+			CollectionType: kmipAdapterType,
+			NumberOfKeys:   1,
+		},
+		Adapters: []KMIPAdapter{
+			newAdapter,
+		},
+	}
+	testAdapters := &KMIPAdapters{
+		Metadata: KeysMetadata{
+			CollectionType: kmipAdapterType,
+			NumberOfKeys:   2,
+		},
+		Adapters: []KMIPAdapter{
+			newAdapter,
+			newAdapter,
+		},
+	}
+	singleCert := &KMIPClientCertificates{
+		Metadata: KeysMetadata{
+			CollectionType: kmipClientCertType,
+			NumberOfKeys:   1,
+		},
+		Certificates: []KMIPClientCertificate{
+			newCertificate,
+		},
+	}
+	testCerts := &KMIPClientCertificates{
+		Metadata: KeysMetadata{
+			CollectionType: kmipClientCertType,
+			NumberOfKeys:   2,
+		},
+		Certificates: []KMIPClientCertificate{
+			newCertificate,
+			newCertificate,
+		},
+	}
+
+	singleKmipObject := &KMIPObjects{
+		Metadata: KeysMetadata{
+			CollectionType: kmipObjectType,
+			NumberOfKeys:   1,
+		},
+		Objects: []KMIPObject{
+			newKmipObject,
+		},
+	}
+	testKmipObjects := &KMIPObjects{
+		Metadata: KeysMetadata{
+			CollectionType: kmipObjectType,
+			NumberOfKeys:   2,
+		},
+		Objects: []KMIPObject{
+			newKmipObject,
+			newKmipObject,
+		},
+	}
```

Comment:
I'm think these should be hardcoded response texts we'd get from the API spec. 

If changes were made to the structures in the SDK that were not in line with the spec, the tests would still technically pass with the way it is structured right now.

Reply:
done.